packages/google_sign_in/google_sign_in_ios/ios/Classes/FLTGoogleSignInPlugin.m

```diff
@@ -116,18 +112,14 @@ - (void)initializeSignInWithParameters:(nonnull FSIInitParams *)params
   if (configuration != nil) {
     self.requestedScopes = [NSSet setWithArray:params.scopes];
     self.configuration = configuration;
-  } else {
-    *error = [FlutterError errorWithCode:@"missing-config"
-                                 message:@"GoogleService-Info.plist file not found and clientId "
-                                         @"was not provided programmatically."
-                                 details:nil];
   }
 }
 
 - (void)signInSilentlyWithCompletion:(nonnull void (^)(FSIUserData *_Nullable,
                                                        FlutterError *_Nullable))completion {
-  [self.signIn restorePreviousSignInWithCallback:^(GIDGoogleUser *user, NSError *error) {
-    [self didSignInForUser:user withCompletion:completion error:error];
+  [self.signIn restorePreviousSignInWithCompletion:^(GIDGoogleUser *_Nullable user,
+                                                     NSError *_Nullable error) {
+    [self didSignInForUser:user serverAuthCode:nil withCompletion:completion error:error];
```

Comment:
You may notice here I set `serverAuthCode` to `nil`. This is something I was unsure about. In previous version, the `serverAuthCode` was a property of the user. However, in the new version, it's now a property of `GIDSignInResult`, which you get when you sign in. I did confirm that in the old version on restore, the `serverAuthCode` was included.

However, I don't think this is a dealbreaker because I think the new way where it doesn't include the `serverAuthCode` is more inline with how OAuth works. Usually that code is a temporary one-time use code and then after that tokens are use to validate (see diagram in https://developers.google.com/identity/sign-in/web/server-side-flow).